src/rabbit_registry.erl

```diff
@@ -114,6 +116,29 @@ conditional_unregister({exchange_decorator, Type}) ->
 conditional_unregister(_) ->
     ok.
 
+% update channels to enable/disable interceptors
+registered({{Class, Type}, ModuleName}) ->
+    ClassModule = class_module(Class),
+    case proplists:get_value(registered, ClassModule:module_info(functions)) of
```

Comment:
Yes, `registered` is not a very specific name. @videlalvaro what functions do you suggest the behaviour would include?